MFVNeutralino/plugins/AnalysisCuts.cc

```diff
@@ -479,379 +589,572 @@ bool MFVAnalysisCuts::filter(edm::Event& event, const edm::EventSetup& setup) {
+  int year = int(MFVNEUTRALINO_YEAR);
+
+  // note that if these weren't pT ordered, we'd have to be more careful in the loops...
+  int njets     = mevent->njets(20);
+  int ncalojets = mevent->calo_jet_pt.size();
+
+  // note that this could be loosened/tightened if desired
+  int sel_btags      = 0;
+  int sel_btags_hard = 0;
+  std::vector<int> calojet_ngood(2, 0); // Two entries (for different pT cuts, each start at 0)
+  std::vector<int> pfjet_ngood(2, 0); // Two entries (for different pT cuts, each start at 0) //FIXME might be temporary
+
+  std::vector<float> jettk_dxys;
+  std::vector<float> jettk_nsigmadxys;
+
+  float alt_calo_ht = 0.0; // To account for the fact that calo_jet_ht uses jets up to eta 5.0. We don't want that
+
+  // FIXME Fix this comment once the code is done
+  for (int ic=0; ic < ncalojets; ic++) {
+    float cj_pt = mevent->calo_jet_pt[ic];
+    float cj_aeta = fabs(mevent->calo_jet_eta[ic]);
+
+    // Do this loop if we want to study correction for JER
+    //if (study_jer) {
+    //    float cj_E  = mevent->calo_jet_energy[ic];
+    //    float closest_pf_dR = 9.9;
+    //    int   closest_pf_idx = 999;
+
+    //    for (int ip=0; ip < njets; ip++) {
+    //        float temp_dR = reco::deltaR(mevent->jet_eta[ip], mevent->jet_phi[ip], mevent->calo_jet_eta[ic], mevent->calo_jet_phi[ic]);
+    //        if (temp_dR < closest_pf_dR) {
+    //            closest_pf_dR = temp_dR;
+    //            closest_pf_idx = ip;
+    //        }
+    //    }
+
+    //    cj_pt = jmt::UncertTools::jer_pt(mevent->jet_gen_energy[closest_pf_idx], cj_E, cj_pt, cj_aeta, jes_jer_var_up);
+
+    //}
+
+    //else if (study_jes) {
+    //    jec_unc.setJetEta(mevent->calo_jet_eta[ic]);
+    //    jec_unc.setJetPt(mevent->calo_jet_pt[ic]);
+    //    if (    jes_jer_var_up) { cj_pt *= (1 + jec_unc.getUncertainty(true)); }
+    //    if (not jes_jer_var_up) { cj_pt *= (1 - jec_unc.getUncertainty(false)); }
+    //}
+
+    if (cj_pt > 30.0 and cj_aeta < 2.5) alt_calo_ht += mevent->calo_jet_pt[ic];
+
+    // Don't do the following CPU-intensive for loop if it's not a relevant jet
+    if (cj_pt < 40.0 or fabs(mevent->calo_jet_eta[ic]) > 2.0) continue;
+
+    // If you made it here, the CaloJet is good enough for the low HT trigger. Record this & check if it's good enough for High HT trig
+    calojet_ngood[0]++;
+    if (cj_pt > 60.0) calojet_ngood[1]++;
+
+  }
+  
+  for(int j0 = 0; j0 < njets; j0++) {
+    float rand_y = distribution(rng);
+    float pf_pt = mevent->jet_pt[j0];
+
+    //if (study_jer) {
+    //    pf_pt = jmt::UncertTools::jer_pt_alt(mevent->jet_gen_energy[j0], mevent->jet_p4(j0), jes_jer_var_up);
+    //}
+
+    //else if (study_jes) {
+    //    jec_unc.setJetEta(mevent->jet_eta[j0]);
+    //    jec_unc.setJetPt(mevent->jet_pt[j0]);
+    //    if (    jes_jer_var_up) { pf_pt *= (1 + jec_unc.getUncertainty(true)); }
+    //    if (not jes_jer_var_up) { pf_pt *= (1 - jec_unc.getUncertainty(false)); }
+    //}
```

Comment:
Do we want all of these JES/JER things to go away? @shaunhogan @pkotamnives @abbywarden